src/main/java/seedu/duke/EditCommand.java

```diff
@@ -0,0 +1,42 @@
+package seedu.duke;
+
+import java.io.IOException;
+import java.util.Date;
+
+public class EditCommand extends Command {
+    protected static String details;
+    protected static int position;
+
+    /**
+     * Constructor.
+     * @param details String containing new Task details.
+     * @param position integer representing position of Task to edit in the taskList
+     */
+    public EditCommand(int position, String details) {
+        this.details = details;
+        this.position = position;
+    }
+
+    @Override
+    public String execute(TaskList t, Ui u, Storage s) {
+        Task tobeEdited = t.list.get(position);
+        String type = tobeEdited.getType();
+        String original = tobeEdited.toString();
+        if (type.equals("T")) {
```

Comment:
Would you consider using a `switch` statement here for the code to be less cluttered?

Reply:
Oh, that's a good idea! Thank you for your suggestion.